homeassistant/components/plex/server.py

```diff
@@ -339,6 +351,22 @@ def connect_to_resource(resource):
             self._plextv_device_cache[client_id] = client
             return client
 
+        for gdm_client in self.hass.data[DOMAIN][GDM_SCANNER].entries:
+            machine_identifier = gdm_client["data"]["Resource-Identifier"]
+            if machine_identifier not in available_clients:
+                baseurl = f"http://{gdm_client['from'][0]}:{gdm_client['data']['Port']}"
+                name = gdm_client["data"]["Name"]
+                try:
+                    device = await self.hass.async_add_executor_job(
```

Comment:
We're doing an `await` inside a loop, which means we are not going to attempt to connect to the next one until the current one is processed. Is that necessary?

Reply:
I think this should happen infrequently enough and quickly enough that it shouldn't matter.

Reply:
A better solution is to extract the iteration and calls to a function and schedule that whole function on the executor.